src/swanctl/swanctl.opt

```diff
@@ -999,6 +1001,9 @@ connections.<conn>.children.<child>.if_id_in = 0
 	ID. The special value _%unique_ sets a unique interface ID on each CHILD_SA
 	instance, beyond that the value _%unique-dir_ assigns a different unique
 	interface ID for each CHILD_SA direction (in/out).
+	This value can be hexadecimal (start with 0x), decimal (no special prefix), or octal (start with 0).
```

Comment:
That's just how any integer values can be configured in these config files. If anything, we should probably add a general section about number/time formats to the strongswan.conf man page (there is a section about time formats in swanctl.conf, which should probably be moved to strongswan.conf, which contains all other information about the general format).

Reply:
Yes, taht would be very very useful.

Reply:
I've pushed a commit with some general information about the format to the _1329-number-formats_ branch (note that we explicitly don't allow octal format to avoid confusion, that `0` prefix was a really weird idea).

Reply:
Well could we use o instead? (Just to accept Octal too. Or maybe explicitely mention that octal is not possible because it's still a common number format for some things)

Reply:
I think it's fine documenting just the supported formats (and decimal and hex should cover practically all use cases). The only place I ever encountered octal numbers is `chmod` (although, considering the range of supported values, they could also be hex). Never related to IKE, IPsec or networking. Do you have an example?

Reply:
No, it's just for completeness. To make sure it's evident, because ineviteably that question will come up if we don't document that it isn't supported.

Reply:
I don't think that's true. It hasn't been supported thus far and nobody ever asked for it (probably because it really isn't common for these use cases). Also, explicitly documenting stuff that isn't supported is tricky because there are always people who read that incorrectly (e.g. miss the _not_), and then assume something is supported because they saw the keyword they expected (and they rarely bother to re-read the documentation if something doesn't work, but immediately complain). So I feel it's better to not bring up octal numbers at all.

Reply:
I agree with you on the point with people not reading texts correctly. I guess it's better to leave it out, as you proposed.